python/packages/autogen-ext/src/autogen_ext/models/_reply_chat_completion_client.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import annotations
+
+import logging
+from typing import Any, AsyncGenerator, List, Mapping, Optional, Sequence, Union
+
+from autogen_core.application.logging import EVENT_LOGGER_NAME
+from autogen_core.base import CancellationToken
+from autogen_core.components.models import (
+    ChatCompletionClient,
+    CreateResult,
+    LLMMessage,
+    ModelCapabilities,
+    RequestUsage,
+)
+from autogen_core.components.tools import Tool, ToolSchema
+
+logger = logging.getLogger(EVENT_LOGGER_NAME)
+
+
+class ReplayChatCompletionClient:
+    """A mock chat completion client that replays predefined responses."""
```

Comment:
We should have doc string here with code blocks examples showing how to use this.  Move the doc string in `__init__` here.